src/pymatgen/io/lobster/outputs.py

```diff
@@ -1677,7 +1677,7 @@ def _read(self, lines: list[str], spin_numbers: list[int]) -> None:
                         _lines.append(float(el))
                 overlaps.append(_lines)
                 if len(overlaps) == len(_lines):
-                    self.band_overlaps_dict[spin]["matrices"].append(np.matrix(overlaps))
```

Comment:
@JaGeo would appreciate it if you could have a look at this as well, got:

```
  /Users/yang/developer/pymatgen/src/pymatgen/io/lobster/outputs.py:1680: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    self.band_overlaps_dict[spin]["matrices"].append(np.matrix(overlaps))
```

And [as per the manual](https://numpy.org/doc/stable/user/numpy-for-matlab-users.html):
> ‘array’ or ‘matrix’? Which should I use?

> Historically, NumPy has provided a special matrix type, np.matrix, which is a subclass of ndarray which makes binary operations linear algebra operations. You may see it used in some existing code instead of np.array. So, which one to use?

> Short answer
> Use arrays.